### PR TITLE
python3: add support for netbsd

### PR DESCRIPTION
**Pull request by Mic92**
## Description of changes

This is needed for https://github.com/NixOS/nix/pull/11566

## Things done

- Built on platform(s)
  - [x] x86_64-linux (pkgsCross.x86_64-freebsd.python3 and pkgsCross.x86_64-netbsd.python3)
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Mic92** [comment]
cc @rhelmot

File: pkgs/development/interpreters/python/cpython/default.nix [hunk 1]
```diff
@@ -307,11 +307,11 @@ in with passthru; stdenv.mkDerivation (finalAttrs: {
     # (since it will do a futile invocation of gcc (!) to find
     # libuuid, slowing down program startup a lot).
     noldconfigPatch
-  ] ++ optionals (stdenv.hostPlatform != stdenv.buildPlatform && stdenv.isFreeBSD) [
+  ] ++ optionals (stdenv.hostPlatform != stdenv.buildPlatform && (stdenv.isFreeBSD || stdenv.hostPlatform.isNetBSD)) [
     # Cross compilation only supports a limited number of "known good"
     # configurations. If you're reading this and it's been a long time
     # since this diff, consider submitting this patch upstream!
-    ./freebsd-cross.patch
```
Comment (Mic92) [inline pkgs/development/interpreters/python/cpython/default.nix:314]: I also made a PR for cpython + freebsd here: https://github.com/python/cpython/pull/124424

**Comment by alyssais** [comment]
Have you tested the NetBSD binary actually works?  Last time I tried this it didn't, which suggests upstream has this disabled for a reason.

**Comment by alyssais** [comment]
Why does Nix need a cross compiled Python anyway?

**Comment by Mic92** [comment]
We cannot build libpsl -> libcurl without it:

```
nix-env % nix build .\#packages.x86_64-linux.nix-x86_64-unknown-netbsd -L --show-trace
error:
       … while calling the 'derivationStrict' builtin
         at <nix/derivation-internal.nix>:34:12:
           33|
           34|   strict = derivationStrict drvAttrs;
             |            ^
           35|

       … while evaluating derivation 'nix-x86_64-unknown-netbsd-2.25.0pre20240924_dirty'
         whose name attribute is located at /nix/store/5n0kgjjm07z3gvz912i5gviig1qh4d4w-source/pkgs/stdenv/generic/make-derivation.nix:336:7

       … while evaluating attribute 'buildInputs' of derivation 'nix-x86_64-unknown-netbsd-2.25.0pre20240924_dirty'
         at /nix/store/5n0kgjjm07z3gvz912i5gviig1qh4d4w-source/pkgs/stdenv/generic/make-derivation.nix:383:7:
          382|       depsHostHost                = elemAt (elemAt dependencies 1) 0;
          383|       buildInputs                 = elemAt (elemAt dependencies 1) 1;
             |       ^
          384|       depsTargetTarget            = elemAt (elemAt dependencies 2) 0;

       … while calling the 'getAttr' builtin
         at <nix/derivation-internal.nix>:44:19:
           43|       value = commonAttrs // {
           44|         outPath = builtins.getAttr outputName strict;
             |                   ^
           45|         drvPath = strict.drvPath;

       … while calling the 'derivationStrict' builtin
         at <nix/derivation-internal.nix>:34:12:
           33|
           34|   strict = derivationStrict drvAttrs;
             |            ^
           35|

       … while evaluating derivation 'curl-x86_64-unknown-netbsd-8.9.1'
         whose name attribute is located at /nix/store/5n0kgjjm07z3gvz912i5gviig1qh4d4w-source/pkgs/stdenv/generic/make-derivation.nix:336:7

       … while evaluating attribute 'propagatedBuildInputs' of derivation 'curl-x86_64-unknown-netbsd-8.9.1'
         at /nix/store/5n0kgjjm07z3gvz912i5gviig1qh4d4w-source/pkgs/stdenv/generic/make-derivation.nix:390:7:
          389|       depsHostHostPropagated      = elemAt (elemAt propagatedDependencies 1) 0;
          390|       propagatedBuildInputs       = elemAt (elemAt propagatedDependencies 1) 1;
             |       ^
          391|       depsTargetTargetPropagated  = elemAt (elemAt propagatedDependencies 2) 0;

       … while calling the 'getAttr' builtin
         at <nix/derivation-internal.nix>:44:19:
           43|       value = commonAttrs // {
           44|         outPath = builtins.getAttr outputName strict;
             |                   ^
           45|         drvPath = strict.drvPath;

       … while calling the 'derivationStrict' builtin
         at <nix/derivation-internal.nix>:34:12:
           33|
           34|   strict = derivationStrict drvAttrs;
             |            ^
           35|

       … while evaluating derivation 'libpsl-x86_64-unknown-netbsd-0.21.5'
         whose name attribute is located at /nix/store/5n0kgjjm07z3gvz912i5gviig1qh4d4w-source/pkgs/stdenv/generic/make-derivation.nix:336:7

       … while evaluating attribute 'buildInputs' of derivation 'libpsl-x86_64-unknown-netbsd-0.21.5'
         at /nix/store/5n0kgjjm07z3gvz912i5gviig1qh4d4w-source/pkgs/stdenv/generic/make-derivation.nix:383:7:
          382|       depsHostHost                = elemAt (elemAt dependencies 1) 0;
          383|       buildInputs                 = elemAt (elemAt dependencies 1) 1;
             |       ^
          384|       depsTargetTarget            = elemAt (elemAt dependencies 2) 0;

       … in the condition of the assert statement
         at /nix/store/5n0kgjjm07z3gvz912i5gviig1qh4d4w-source/lib/customisation.nix:352:23:
          351|             drvPath = assert condition; drv.${outputName}.drvPath;
          352|             outPath = assert condition; drv.${outputName}.outPath;
             |                       ^
          353|           } //

       … while evaluating the attribute 'handled'
         at /nix/store/5n0kgjjm07z3gvz912i5gviig1qh4d4w-source/pkgs/stdenv/generic/check-meta.nix:507:7:
          506|       # or, alternatively, just output a warning message.
          507|       handled =
             |       ^
          508|         (

       … from call site
         at /nix/store/5n0kgjjm07z3gvz912i5gviig1qh4d4w-source/pkgs/stdenv/generic/check-meta.nix:511:13:
          510|           else if valid == "no" then (
          511|             handleEvalIssue { inherit meta attrs; } { inherit (validity) reason errormsg; }
             |             ^
          512|           )

       … while calling 'handleEvalIssue'
         at /nix/store/5n0kgjjm07z3gvz912i5gviig1qh4d4w-source/pkgs/stdenv/generic/check-meta.nix:269:38:
          268|
          269|   handleEvalIssue = { meta, attrs }: { reason , errormsg ? "" }:
             |                                      ^
          270|     let

       … while calling the 'throw' builtin
         at /nix/store/5n0kgjjm07z3gvz912i5gviig1qh4d4w-source/pkgs/stdenv/generic/check-meta.nix:281:8:
          280|         else throw;
          281|     in handler msg;
             |        ^
          282|

       error: Package ‘python3-3.12.5’ in /nix/store/5n0kgjjm07z3gvz912i5gviig1qh4d4w-source/pkgs/development/interpreters/python/cpython/default.nix:675 is not available on the requested hostPlatform:
         hostPlatform.config = "x86_64-unknown-netbsd"
         package.meta.platforms = [
           "aarch64-linux"
           "armv5tel-linux"
           "armv6l-linux"
           "armv7a-linux"
           "armv7l-linux"
           "i686-linux"
           "loongarch64-linux"
           "m68k-linux"
           "microblaze-linux"
           "microblazeel-linux"
           "mips-linux"
           "mips64-linux"
           "mips64el-linux"
           "mipsel-linux"
           "powerpc64-linux"
           "powerpc64le-linux"
           "riscv32-linux"
           "riscv64-linux"
           "s390-linux"
           "s390x-linux"
           "x86_64-linux"
           "x86_64-darwin"
           "i686-darwin"
           "aarch64-darwin"
           "armv7a-darwin"
           "i686-cygwin"
           "x86_64-cygwin"
           "x86_64-windows"
           "i686-windows"
           "i686-freebsd"
           "x86_64-freebsd"
         ]
         package.meta.badPlatforms = [ ]
       , refusing to evaluate.

       a) To temporarily allow packages that are unsupported for this system, you can use an environment variable
          for a single invocation of the nix tools.

            $ export NIXPKGS_ALLOW_UNSUPPORTED_SYSTEM=1

          Note: When using `nix shell`, `nix build`, `nix develop`, etc with a flake,
                then pass `--impure` in order to allow use of environment variables.

       b) For `nixos-rebuild` you can set
         { nixpkgs.config.allowUnsupportedSystem = true; }
       in configuration.nix to override this.

       c) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
         { allowUnsupportedSystem = true; }
       to ~/.config/nixpkgs/config.nix.
```

I haven't tested the actual interpreter because it's not actually used in nix.
However if basic things like python already don't cross compile than we might need to disable netbsd cross compilation for nix.

**Comment by Mic92** [comment]
It's the second time this `psl-make-dafsa` script makes life as a cross-compiler hard...
I think it would be overall useful not depend on python for libcurl and have the script package separately.

**Comment by alyssais** [comment]
Before we've talked about not including the Python parts of libpsl when we can't cross compile Python.  Can't we do that?

**Comment by Mic92** [comment]
https://github.com/NixOS/nixpkgs/pull/344209

**Comment by Mic92** [comment]
Here it's just disabled for cross-compilation: https://github.com/NixOS/nixpkgs/pull/344211

**Comment by Mic92** [comment]
Opened an issue instead: https://github.com/NixOS/nixpkgs/issues/344423

**Comment by alyssais** [comment]
> Sure, but I actually, I don't want this dependency at all in libcurl. As far as I can see, no package really needs it.

Why not just disable it universally then?

**Comment by Mic92** [comment]
I did this here: https://github.com/NixOS/nixpkgs/pull/344209

**Comment by alyssais** [comment]
I don't understand from that what it's needed for.

**Pull request closed without merging**